data_model/src/metadata.rs

```diff
@@ -122,6 +128,11 @@ impl Metadata {
         map.get(key)
     }
 
+    /// Return if the metadata map contains the given key.
+    pub fn contains(&self, key: &Name) -> bool {
+        self.map.contains_key(key)
+    }
+
     /// Returns iterator over key - value pairs stored in [`Metadata`]
```

Comment:
```suggestion
    /// Return the iterator over key-value pairs stored in [`Metadata`].
```

Reply:
just a note: I've changed `returns` to `return` here just for consistency based on other files in this PR, but probably the 3rd person is more appropriate and it's better to make the change in other places instead (e.g. `construct` -> `constructs`)

I'm not sure what is more prevalent in your code base, I'm noticing the use of both verb forms, and we'd better be consistent

Reply:
No. Functions do things ⇒ their docstrings should be imperative. This is the convention in Python, Lisp and the Linux kernel.